test/cloudtest/test_storaged_shared_fate.py

```diff
@@ -0,0 +1,219 @@
+        f"""
+            > CREATE MATERIALIZED VIEW v{i} AS SELECT COUNT(*) FROM t1 UNION ALL SELECT COUNT(*) FROM s{i};
+    """
+        for i in range(NUM_SOURCES)
+    )
+
+    create_indexes_and_counts = "\n".join(
+        f"""
+            > CREATE DEFAULT INDEX ON v{i};
+
+            > SELECT COUNT(*) > 0 FROM s{i};
+            true
+    """
+        for i in range(NUM_SOURCES)
+    )
+
+    mz.testdrive.run(
+        # TODO: I need both clusters? Can't create index/mv in cluster which has sources connected with IN CLUSTER
+        input=dedent(
+            f"""
+            > CREATE CLUSTER storaged_shared_fate REPLICAS (storaged_shared_fate_replica (SIZE '{CLUSTER_SIZE}-1'));
+
+            > CREATE CLUSTER shared_fate REPLICAS (shared_fate_replica (SIZE '{CLUSTER_SIZE}-1'));
+
+            > SET cluster = storaged_shared_fate;
+
+            > CREATE CONNECTION kafka TO KAFKA (BROKER '${{testdrive.kafka-addr}}')
+
+            $ kafka-create-topic topic=storaged-shared-fate partitions={CLUSTER_SIZE}
+
+            $ kafka-ingest format=bytes topic=storaged-shared-fate repeat=1000
+            CDE${{kafka-ingest.iteration}}
+
+            {create_sources}
+
+            > CREATE TABLE t1 (f1 INTEGER);
+
+            > INSERT INTO t1 SELECT 123000 + generate_series FROM generate_series(1, 1000);
+
+            > INSERT INTO t1 SELECT 234000 + generate_series FROM generate_series(1, 1000);
+
+            > SET cluster = shared_fate;
+
+            {create_views}
+
+            > SET cluster = storaged_shared_fate;
+
+            $ kafka-ingest format=bytes topic=storaged-shared-fate repeat=1000
+            DEF${{kafka-ingest.iteration}}
+
+            > SET cluster = shared_fate;
+
+            {create_indexes_and_counts}
+            """
+        ),
+        seed=seed,
+    )
+
+
+def validate(mz: MaterializeApplication, seed: int) -> None:
+    source_view_validations = "\n".join(
+        f"""
+            > SELECT COUNT(*) FROM s{i};
+            3000
+
+            > SELECT * FROM v{i};
+            3000
+            3000
+    """
+        for i in range(NUM_SOURCES)
+    )
+
+    mz.testdrive.run(
+        input=dedent(
+            f"""
+            > SET cluster = storaged_shared_fate;
+
+            > INSERT INTO t1 SELECT 345000 + generate_series FROM generate_series(1, 1000);
+
+            $ kafka-ingest format=bytes topic=storaged-shared-fate repeat=1000
+            EFG${{kafka-ingest.iteration}}
+
+            > SET cluster = shared_fate;
+
+            {source_view_validations}
+
+            > DROP CLUSTER storaged_shared_fate CASCADE;
+
+            > DROP CLUSTER shared_fate CASCADE;
+            """
+        ),
+        no_reset=True,
+        seed=seed,
+    )
+
+
+def kill_clusterd(
+    mz: MaterializeApplication, compute_id: int, signal: str = "SIGKILL"
+) -> None:
+    cluster_id, replica_id = mz.environmentd.sql_query(
+        # TODO: Can't find storaged_shared_fate_replica, is that expected?
```

Comment:
I think this is a problem, why can't I find the storaged replica to kill them?

Reply:
one sec let me check what is going on

Reply:
I think I know what is going on -- the `test_*` names  in this files are identical to the ones from the file you copied -- so if you use `./pytest -k test_xxx` to run a test, it actually ran 2 tests, one from each py file and the one from the other file did not have your new storage cluster.

With this patch:

```
diff --git a/test/cloudtest/test_storaged_shared_fate.py b/test/cloudtest/test_storaged_shared_fate.py
index 0e1bdc3f6..4eaf75be9 100644
--- a/test/cloudtest/test_storaged_shared_fate.py
+++ b/test/cloudtest/test_storaged_shared_fate.py
@@ -131,7 +131,7 @@ def kill_clusterd(
 ) -> None:
     cluster_id, replica_id = mz.environmentd.sql_query(
         # TODO: Can't find storaged_shared_fate_replica, is that expected?
-        f"SELECT cluster_id, id FROM mz_cluster_replicas WHERE name = 'shared_fate_replica'"
+        f"SELECT cluster_id, id FROM mz_cluster_replicas WHERE name = 'storaged_shared_fate_replica'"
     )[0]
 
     pod_name = f"pod/cluster-{cluster_id}-replica-{replica_id}-{compute_id}"
@@ -147,7 +147,7 @@ def kill_clusterd(
         pass
 
 
-def test_kill_all_clusterds(mz: MaterializeApplication) -> None:
+def test_kill_all_storage_clusterds(mz: MaterializeApplication) -> None:
     """Kill all clusterds"""
     populate(mz, 1)

```

this command:

```
./pytest -k test_kill_all_storage_clusterds
```
runs to completion, passes, and seems to be killing and testing the right thing.